subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -333,10 +341,12 @@ STATIC int build_msg_block_for_send(struct lwm2m_message *msg, uint16_t block_nu
 			return ret;
 		}
 		ret = coap_block_transfer_init(msg->out.block_ctx, lwm2m_default_block_size(),
-					       msg->body_encode_buffer.offset);
+					       complete_payload_len);
 		if (ret < 0) {
 			return ret;
 		}
+		sys_slist_append(&ongoing_block_sends, &msg->node);
```

Comment:
We're effectively pushing the same `sys_snode_t msg->node` to two lists, `ongoing_block_sends` here and `pending_sends` [there](https://github.com/SeppoTakalo/zephyr/blob/lwm2m_opaque_get/subsys/net/lib/lwm2m/lwm2m_message_handling.c#L700.). This doesn't seem correct.
Since we only support a single outgoing BLOCK2 transfer, perhaps just store the pointer (exclusively for BLOCK2 transfers)?